### ISSUE TITLE
How to get the first 500 most-common user agents

### ISSUE THREAD

**Issue by LeMoussel** [intoli/user-agents#3]
Hi,

I do this `userAgent = new UserAgents({ deviceCategory: 'desktop', userAgent: /^Mozilla/ });` to get Mozilla user agent on desktop device.

I want to get the first 500 most-common/recent Mozilla user agent on desktop device.
To do that I probably need to do the filtering using the [custom function](https://github.com/intoli/user-agents#custom-filter-functions).

I don't know how to filter to get the first 500 most-common/recent Mozilla user agent on desktop device.

Can you help me?

**Comment by sangaline** [comment]
This isn't supported by the current API, but it's a good feature suggestion.

**Comment by LeMoussel** [comment]
I see your blog post [User-Agents — A random user agent generation library that's always up to date](https://intoli.com/blog/user-agents/) but I couldn't find any information about the UA file description.

In the file, does the UAs are sorted in the file based on the weight?

**Comment by sangaline** [comment]
Yes, they are sorted based on weight. If you would like to parse the `user-agents.json.gz` file directly, then you can safely assume that they are sorted by decreasing weight. The weight is also included in each JSON entry.

**Comment by kopollo** [comment]
https://whatismy.tech/ is giving you popular User Agent list on global scale.
